DEVELOPERS.md

```diff
@@ -0,0 +1,31 @@
+# Developers Notes
+
+## Setup
+
+- Please use the `make git_hooks` to setup the pre-commit hooks
+
+## Linting
+
+At the moment we are using pyright pegged at version 1.1.294
+
+## VS Code
+
+Probably fair to say most of use VS Code so its worth writing something here
+about our shared setup
+
+### Linting
+
+We use pyright for linting (see .pre-commit-config.yml) but [pyright recommend
+that you use Pylance when using VS Code](https://microsoft.github.io/pyright/#/installation?id=vs-code).
+
+Here is a snippet of the recommended:
+
+```
+    "python.analysis.diagnosticMode": "workspace",
+    "python.analysis.typeCheckingMode": "basic",
+    "python.analysis.include": [
+        "**/*.py"
+    ],
+    "editor.defaultFormatter": "ms-python.black-formatter",
+    "python.languageServer": "Pylance",
+```
```

Comment:
I personally add this as well to my User Settings, which creates two vertical lines in my VScode for Python files only - one at the 72 column (PEP8: max docstrings/comments line length 72) and one at the 88 column (max code line length according to CPR's engineering standards & guidelines):

```
"workbench.colorCustomizations": {"editorRuler.foreground": "#63636331"},
"[python]": {
        "editor.rulers": [72, 88]
    },
```

Reply:
That's a good idea - maybe one for your own PR ... a new section "Useful (Optional) Settings" ?